hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/plan/generators/BaseHoodieCompactionPlanGenerator.java

```diff
@@ -182,12 +186,14 @@ public HoodieCompactionPlan generateCompactionPlan(String compactionInstant) thr
     return compactionPlan;
   }
 
-  protected abstract HoodieCompactionPlan getCompactionPlan(HoodieTableMetaClient metaClient, List<HoodieCompactionOperation> operations);
+  protected abstract List<String> getPartitions();
+
+  protected abstract HoodieCompactionPlan getCompactionPlan(HoodieTableMetaClient metaClient, List<HoodieCompactionOperation> operations, Pair<List<String>,List<String>> partitionPair);
```

Comment:
Do we need the missing partitions to generate the plan?

Reply:
because we need to record this missing partitions(`.setMissingSchedulePartitions(res)`) in plan during `getCompactionPlan function`